src/domain/event/similarEvents/__tests__/SimilarEvents.test.tsx

```diff
@@ -83,3 +98,30 @@ test('should render similar event cards', async () => {
     ).toBeInTheDocument();
   });
 });
+
+it('has return path on similar event link', async () => {
+  const path = ROUTES.EVENT;
+  const route = path.replace(':id', event.id);
+  const { history } = await waitForComponentToBeLoaded(() =>
+    renderWithRoute(<SimilarEvents event={event} />, {
+      mocks,
+      path,
+      routes: [route],
+    })
+  );
+  expectedSimilarEvents.data.forEach((similarEvent) => {
+    userEvent.click(
+      screen.queryByRole('button', {
+        name: translations.event.eventCard.ariaLabelLink.replace(
+          '{{name}}',
+          similarEvent.name.fi
+        ),
+      })
+    );
+    expect(history.location).toMatchObject({
+      pathname: `/fi${ROUTES.EVENT.replace(':id', similarEvent.id)}`,
+      search: `?returnPath=${encodeURIComponent(route)}`,
+    });
+    history.goBack();
```

Comment:
Are you aware that this works without further waiting because data comes instantly from cache (not async)?

for example if I change EventListQuery to:

```
  const { data: eventsData, loading } = useEventListQuery({
    ssr: false,
    variables: eventFilters,
    fetchPolicy: 'network-only',
  });
```

this breaks

Reply:
I would try to make this test more resilient by adding wating functionality to every iteration of this for loop

Reply:
Good point, not sure did I do it now correctly, I added the loadingspinner wait inside the loop and the test wanted it also to wrapped with `act` but now it passed with that network-only policy

Reply:
You haven't pushed any changes yet?

Reply:
now pushed, the fixing the last comment took some time!

Reply:
I was just making sure because that one time you forgot to push :D